code/src/layouts/header/components/header-search-input/HeaderSearchInput.test.tsx

```diff
@@ -1,40 +1,187 @@
+
+      const dropdown = screen.queryByTestId("search-dropdown");
+      expect(dropdown).not.toBeInTheDocument();
+    });
+
+    test("opens dropdown when search query length is 4 or more", async () => {
+      const searchField = screen.getByPlaceholderText(
+        /header.searchInputPlaceholder/
+      );
+      expect(searchField).toBeInTheDocument();
+
+      await typeIntoInput(searchField, "test");
+      expect(searchField).toHaveValue("test");
+
+      const dropdown = await screen.findByTestId("search-dropdown");
+      expect(dropdown).toBeInTheDocument();
+    });
+
+    test("does not open dropdown when search query length is less than 4", async () => {
+      const searchField = screen.getByPlaceholderText(
+        /header.searchInputPlaceholder/
+      );
+      expect(searchField).toBeInTheDocument();
+
+      await typeIntoInput(searchField, "tes");
+      expect(searchField).toHaveValue("tes");
+
+      const dropdown = screen.queryByTestId("search-dropdown");
+      expect(dropdown).not.toBeInTheDocument();
+    });
   });
 
-  test("changes input value", async () => {
-    const searchField = screen.getByPlaceholderText("Search...");
-    expect(searchField).toBeInTheDocument();
+  describe("Without results", () => {
+    test("displays loading label when isLoading is true", async () => {
+      (useGetUserProductsBySearchQuery as jest.Mock).mockReturnValue({
+        data: null,
+        isLoading: true,
+        isError: false
+      });
+
+      renderWithProviders(<HeaderSearchInput />);
+      const searchField = screen.getByPlaceholderText(
+        /header.searchInputPlaceholder/
+      );
+      expect(searchField).toBeInTheDocument();
+
+      await typeIntoInput(searchField, "test");
+      expect(searchField).toHaveValue("test");
+
+      const loadingLabelElement = await screen.findByText(loadingLabel);
+      expect(loadingLabelElement).toBeInTheDocument();
+    });
+
+    test("displays no results label when totalElements is 0 and not loading", async () => {
+      (useGetUserProductsBySearchQuery as jest.Mock).mockReturnValue({
+        data: {
+          totalElements: 0,
+          content: []
+        },
+        isLoading: false,
+        isError: false
+      });
+
+      renderWithProviders(<HeaderSearchInput />);
+      const searchField = screen.getByPlaceholderText(
+        /header.searchInputPlaceholder/
+      );
+      expect(searchField).toBeInTheDocument();
 
-    await typeIntoInput(searchField, "test");
+      await typeIntoInput(searchField, "test");
 
-    expect(searchField).toHaveValue("test");
+      const noResultsLabelElement = await screen.findByText(noResultsLabel);
+      expect(noResultsLabelElement).toBeInTheDocument();
+    });
   });
 
-  test("clears input value when clear button is clicked", async () => {
-    const searchField = screen.getByPlaceholderText("Search...");
-    expect(searchField).toBeInTheDocument();
+  describe("Query Params", () => {
```

Comment:
The name of `describe` block should be the same as file name for convenience

Reply:
Thank you, renamed